### ISSUE TITLE
Wouldn't start on Debian 11

### ISSUE THREAD

**Issue by virus2500** [deexno/checkmk-telegram-plus#2]
Hi, 

thanks for the extension. Just one minor problem occured to me during the install: The bot simply wouldn't start via systemd.

```
Jun 21 23:24:27 checkmk systemd[1523085]: checkmk-telegram-plus-main.service: Failed to locate executable /usr/sbin/runuser: No such file or directory
Jun 21 23:24:27 checkmk systemd[1523085]: checkmk-telegram-plus-main.service: Failed at step EXEC spawning /usr/sbin/runuser: No such file or directory
```

Now in Debian 11 runuser is in /sbin

```
┌─[✗]─[root@checkmk]-[/etc/systemd/system]
└─╼ # which runuser
/sbin/runuser
```

I changed my unit file to this
```
[Unit]
Description=CheckMK Telegram Plus Service

[Service]
# Wait for the OMD Page to fully start
ExecStartPre=/bin/sleep 60
User=main
WorkingDirectory=/omd/sites/main/local/share/checkmk-telegram-plus
# The service must be started as follows, as required environment variables are not set if the service is only executed in the name of this user.
ExecStart=/omd/sites/main/bin/python3 /omd/sites/main/local/share/checkmk-telegram-plus/telegram_bot.py
Restart=always

[Install]
WantedBy=multi-user.target
```

I have read the message "The service must be started as follows, as required environment variables are not set if the service is only executed in the name of this user." but couldn't find (yet) anything that isn't working so... i don't know if this would impact other systems.

Maybe you could even remove the need for ExecStart=/omd/sites/main/bin/python3 with an shebang in telegram_bot.py since which as the user already points to the right python version?

```
OMD[main]:~$ which python3
/omd/sites/main/bin/python3
```

thanks again!

**Comment by deexno** [comment]
Hi, I'm glad to hear you're using my Telegram bot.

In advance, the problem has nothing to do with Debian 11, I myself have 2 installations running under Debian 11 without problems. (Just btw. because of the title)

I would advise against removing /sbin/runuser! This will lose the function "🔄 RESCHEDULE CHECK" as well as possible future functions.

All other functions, you are right, should work.

I also think I already found the issue, in the service it tries to run runuser under "/usr/sbin/runuser" but you installed it under "/sbin/runuser". 

Try it this way:
ExecStart=/sbin/runuser -l mhs -c 'cd /omd/sites/mhs/local/share/main && /omd/sites/mhs/bin/python3 /omd/sites/mhs/local/share/main/telegram_bot.py'

If this works, I will extend the install script so that it checks where runuser is installed beforehand. I still don't understand why apt decided to install it here but okay.

**Comment by virus2500** [comment]
Hi,

thanks for the reply. I think i now know why it is in /sbin instead of /usr/sbin. All my VMs (lxc) have it installed in /sbin but all my physical hosts and qemu VMs) have it in /usr/sbin. 

I was using the Debian 11 standard lxc template (provided via proxmox). I even fired up a new VM and its again just in /sbin. 
Weird that there is a difference.

I slightly changed to Exec for my site
```
ExecStart=/sbin/runuser -l main -c 'cd /omd/sites/main/local/share/checkmk-telegram-plus/ && /omd/sites/main/bin/python3 /omd/sites/main/local/share/checkmk-telegram-plus/telegram_bot.py'
```
And yep, reschedule is working now. I thought it was working because even with the other line the bot replied with
```
RESCHEDULE CHECK WAS COMPLETED SUCCESSFULLY
```

but apparently it wasn't doing anything because now i also get the agent ([agent] Sucess etc.) data which wasn't there before.

To summarize: A check for the location of runuser during the install would be awesome 😄 

Another thing i just found... (maybe i am just using it wrong)
1) hit reschedule check
2) select a hostgroup (and missclick on a group you don't want)
3) type /cancel
    bot replies with "Conversation cancelled X"
4) hit reschedule again
    bot replies (without letting you select a hostgroup etc)

```
Failed to lookup IPv4 address of 🔄 RESCHEDULE CHECK via DNS: [Errno -2] Name or service not known


RESCHEDULE CHECK WAS COMPLETED SUCCESSFULLY
```

I don't know if this is me using it wrong (missunderstanding the /cancel command) or really a bug.

Anyway, thanks for the extension again. Loving it so far 👍 😄

**Comment by deexno** [comment]
I'm glad to hear that it now works this way. I will modify the installation script accordingly. As soon as the problem is solved, I will report here again for possible future users and close this post. 

Regarding the issue with /cancel, I still have to check. Actually, cancel should cancel the entire conversation, but apparently the bot continues to interpret your requests. I'll have a look at it. Can you please open another post for this problem? Thanks!

And I'm glad you like the bot, by the way I'm always open for new features, so feel free to create a feature request here if you still need certain functions. As far as it should be possible for me to implement them, I will do it! 😄

**Comment by virus2500** [comment]
Opened #3 for the cancel issue!

Thanks again for updating it and being open for feature requests etc 👍 😄

**Comment by deexno** [comment]
The problem should now be solved. I have tested it from my side, and it works as planned. So in the future, it should not matter where runuser is installed. 😄
